Lib/test/test_typing.py

```diff
@@ -6350,6 +6352,12 @@ def test_deprecation_for_no_type_params_passed_to__evaluate(self):
 
         self.assertEqual(cm.filename, __file__)
 
+    def test_collect_parameters(self):
+        typing = import_helper.import_fresh_module("typing")
+        with self.assertWarns(DeprecationWarning) as cm:
```

Comment:
I would probably do this just to make sure that it is actually _this_ function being deprecated, not some other function:

```suggestion
        with self.assertWarnsRegex(
            DeprecationWarning,
            "The private _collect_parameters function is deprecated"
        ) as cm:
```